library/src/test/java/com/novoda/downloadmanager/DownloadManagerTest.java

```diff
@@ -0,0 +1,246 @@
+    public void downloadGivenBatch() {
+        downloadManager.download(BATCH);
+
+        verify(downloadManagerDownloader).download(BATCH, downloadBatches);
+    }
+
+    @Test
+    public void doesNotPause_whenBatchIdIsUnknown() {
+        downloadManager.pause(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void pausesBatch() {
+        downloadManager.pause(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).pause();
+    }
+
+    @Test
+    public void doesNotResume_whenBatchIdIsUnknown() {
+        downloadManager.pause(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void doesNotResume_whenBatchIsAlreadyDownloading() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().withStatus(DownloadBatchStatus.Status.DOWNLOADING).build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        InOrder inOrder = inOrder(downloadBatch);
+        inOrder.verify(downloadBatch).status();
+        inOrder.verifyNoMoreInteractions();
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenResuming() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void resumesBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).resume();
+    }
+
+    @Test
+    public void triggersDownload_whenResumingBatch() {
+        given(downloadBatch.status()).willReturn(anInternalDownloadsBatchStatus().build());
+
+        downloadManager.resume(DOWNLOAD_BATCH_ID);
+
+        verify(downloadManagerDownloader).download(downloadBatch, downloadBatches);
+    }
+
+    @Test
+    public void doesNotDelete_whenBatchIdIsUnknown() {
+        downloadManager.delete(new LiteDownloadBatchId("unknown"));
+
+        verifyZeroInteractions(downloadBatch, additionalDownloadBatch);
+    }
+
+    @Test
+    public void removesBatchFromInternalList_whenDeleting() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        assertThat(downloadBatches).doesNotContainEntry(DOWNLOAD_BATCH_ID, downloadBatch);
+    }
+
+    @Test
+    public void deletesBatch() {
+        downloadManager.delete(DOWNLOAD_BATCH_ID);
+
+        verify(downloadBatch).delete();
+    }
+
+    @Test
+    public void addsCallbackToInternalList() {
+        DownloadBatchCallback additionalDownloadBatchCallback = mock(DownloadBatchCallback.class);
+
+        downloadManager.addDownloadBatchCallback(additionalDownloadBatchCallback);
+
+        assertThat(downloadBatchCallbacks).contains(additionalDownloadBatchCallback);
```

Comment:
We could test this by doing an action on `downloadManager` and verify the response is received rather than expose the internals of the class